low-level/sensors/impl/generic/node_hw.py

```diff
@@ -1170,41 +1172,63 @@ def _parse_psu_unit_info(self, index, date, _time, sensor, sensor_num, event, st
         self._send_json_msg(resource_type, alert_type, severity, info, specific_info)
         store.put(self.faulty_resources, self.faulty_resources_path)
 
+    def _get_sensor_num_by_id(self, hw_type, sensor_num):
+        """Assign and return sensor index to sensors of given hw_type.
+
+           Usecase:
+            sensor_id_map = {
+                "Power Supply": {
+                    'f1': 'PS Redundancy',
+                    'f2': 'Status',
+                    'f3': 'Status'
+                    }
+                }
+            In such case where sensor_id's are not unique or does not
+            have a numerical identifier, assign numerical sensor id
+            starting with 0.
+            indices are assigned according to sorted hexadecimal
+            sensor number.
+        """
+        return sorted(list(
+            self.sensor_id_map[hw_type].keys())).index(sensor_num)
+
     def _parse_disk_info(self, index, date, _time, sensor, sensor_num, event, status, is_last):
         """Parse out Disk related changes that gets reaflected in the ipmi sel list"""
 
         sensor_id = self.sensor_id_map[self.TYPE_DISK][sensor_num]
+        disk_slot = re.search(r'\d+', sensor_id)
+        if disk_slot:
+            disk_slot = disk_slot.group()
+        else:
+            disk_slot = self._get_sensor_num_by_id(self.TYPE_DISK, sensor_num)
+        if 'Status' in sensor_id:
+            disk_name = sensor_id.replace('Status', f'(0x{sensor_num})')
+        else:
+            disk_name = sensor_id +  f' (0x{sensor_num})'
 
         common, specific, specific_dynamic = self._get_sensor_props(sensor_id)
         if common:
-            disk_sensors_list = self._get_sensor_list_by_entity(common['Entity ID'])
-            disk_sensors_list.remove(sensor_id)
-
-            description = f"{event} - {status}"
-
             if not specific:
                 specific = {"States Asserted": "N/A", "Sensor Type (Discrete)": "N/A"}
             specific_info = specific
-            alert_severity_dict = {
-                ("Drive Present", "Asserted"): ("insertion", "informational"),
-                ("Drive Present", "Deasserted"): ("missing", "critical"),
-                }
             resource_type = NodeDataMsgHandler.IPMI_RESOURCE_TYPE_DISK
-            fru = self.ipmi_client.is_fru(self.fru_map[self.TYPE_DISK])
+            try:
+                alert = alert_for_event[self.TYPE_DISK][event][status]
+            except KeyError:
+                alert = Alert("fault", "informational", f"{event} - {status}",
+                              "unknown", DEFAULT_RECOMMENDATION)
```

Comment:
I observed different behavior for the situation when the event (eg. "Drive Fault") is not present in our known list of events, For *Disks*, we Raise a `fault` alert with severity `informational` but in *Power Supply* we `log the error and return` back.
Which one of the approaches is better?

Reply:
>> For Disks, we Raise a fault alert with severity informational
Generally for fault we define severity as 'error/critical'.

For unknown event we should log the error and return back, because with current approach for 
 "unknown_event asserted" or "unknown_event deasserted" for both case we will raise fault alert r8??

Reply:
Yes Addressed 👍 . I feel the same we should not raise an alert for an unknown event.